implementations/rust/ockam/ockam_api/src/cloud/project/controller_client.rs

```diff
@@ -72,6 +74,126 @@ impl ControllerClient {
             .miette_success("get orchestrator version")
     }
 
+    pub async fn add_space_admin(
+        &self,
+        ctx: &Context,
+        space_id: &str,
+        email: &EmailAddress,
+    ) -> miette::Result<AdminInfo> {
+        trace!(target: TARGET, "adding new space administrator");
+        let admins: Vec<AdminInfo> = self
+            .get_secure_client()
+            .ask(
+                ctx,
+                "spaces",
+                Request::put(format!("/v0/{space_id}/admins/{email}")),
+            )
+            .await
+            .into_diagnostic()?
+            .miette_success("add space admins")?;
+        admins
+            .into_iter()
+            .find(|a| a.email == email.to_string())
+            .ok_or(miette!("The email {email} was not added to the space"))
```

Comment:
The orchestrator returns the full list of admins in all the endpoints. Changing that is out of scope for me right now, so that's why I added this "normalization" step in the client to return 1/n/empty depending on the endpoint. 

In this case, since the orchestrator returns the whole list, this error shouldn't happen.

Reply:
@ckhrysze can you please make a note somewhere to improve this? Does the current API make sense to you?